pages/library/index.jsx

```diff
@@ -5,29 +5,99 @@ import styled from "styled-components";
 import Pagination from "@/components/Pagination";
 import MenuButton from "@/components/buttons/MenuButton";
 import MenuOption from "@/components/MenuOption";
+import SearchSvg from "@/public/search.svg";
 
 const fetcher = (...args) => fetch(...args).then((response) => response.json());
 
 export default function Library({ menuMode, setMenuMode }) {
   const [page, setPage] = useState(1);
+  const [searchTerm, setSearchTerm] = useState("");
+  const [searchResults, setSearchResults] = useState(null);
+  const [loading, setLoading] = useState(false);
 
   useEffect(() => {
     setMenuMode("closed");
   }, [setMenuMode]);
 
-  const { data, error, isLoading } = useSWR(`/api/${page}`, fetcher);
+  async function searchGame(searchTerm, page) {
+    setLoading(true);
+    try {
+      const searchResponse = await fetch(`api/${page}`, {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({ searchTerm }),
+      });
 
-  if (isLoading) return <div>Loading...</div>;
+      if (!searchResponse.ok) {
+        throw new Error(`Error: ${searchResponse.status}`);
+      }
+
+      const searchData = await searchResponse.json();
+      return searchData;
+    } catch (error) {
+      console.error("Error during game search:", error);
+    } finally {
+      setLoading(false);
+    }
+  }
+
+  async function handleSearch(event) {
+    event.preventDefault();
+    const formData = new FormData(event.target);
+    let searchFormData = Object.fromEntries(formData);
+    setSearchTerm(searchFormData.searchInput);
+    setPage(1);
+    setSearchResults(null);
+  }
+
+  useEffect(() => {
+    async function fetchData() {
+      if (searchTerm) {
+        const results = await searchGame(searchTerm, page);
+        setSearchResults(results);
+      }
+    }
+    if (searchTerm) {
+      fetchData();
+    }
+  }, [searchTerm, page]);
+
+  const { data, error } = useSWR(searchTerm ? null : `/api/${page}`, fetcher);
+
+  if (loading || (!searchTerm && !data)) {
+    return <div>Loading...</div>;
+  }
 
   if (error) return <div>Error: {error.message}</div>;
 
+  const displayData = searchResults || data;
+
   return (
     <>
       <h1>Library</h1>
-      {data ? (
+      <StyledForm onSubmit={handleSearch}>
+        <StyledSearchInput
+          type="text"
+          aria-label="search input"
+          placeholder="Type your search"
+          name="searchInput"
+          required
+        />
+        <StyledSubmitButton type="submit">
+          <SearchSvg />
+        </StyledSubmitButton>
+      </StyledForm>
+      {searchTerm && (
+        <StyledSearchTerm>
+          You searched for: <strong>{searchTerm}</strong>
+        </StyledSearchTerm>
+      )}
+      {displayData && displayData.length > 0 ? (
```

Comment:
you can shorten this:
```suggestion
      {displayData?.length > 0 ? (
```